src/server/express.ts

```diff
@@ -133,3 +135,35 @@ async function methodNotAllowed(_req: Request, res: Response): Promise<void> {
     }),
   );
 }
+
+async function generateJwt(req: Request, res: Response): Promise<void> {
+  const { connectedAppClientId, connectedAppSecretId, connectedAppSecretValue } = getConfig();
+
+  const { username, scopes, source, resource, server, siteName } = req.body;
+  if (!username || !scopes || !source || !resource || !server || !siteName) {
+    res.status(400).json({
+      error: 'username, scopes, source, resource, server, and siteName are required',
+    });
+    return;
+  }
+
+  const additionalPayload: Record<string, unknown> = {};
+  if (resource === 'query-datasource') {
+    additionalPayload.region = 'West';
+  }
```

Comment:
This is a simple example of how a customer could customize the user attributes based on any of the input params, including the username, Tableau server URL, Tableau site, and MCP tool being called.